tests/integration/api/test_api.py

```diff
@@ -407,31 +252,26 @@ def test_graph_snapshot(self):
 
         # Test without an event query
         snap = dog.Graph.create(metric_query=metric_query, start=start, end=end)
-        ok('snapshot_url' in snap, msg=snap)
-        ok('metric_query' in snap, msg=snap)
-        ok('event_query' not in snap, msg=snap)
-        eq(snap['metric_query'], metric_query)
-        snapshot_url = snap['snapshot_url']
-        while dog.Graph.status(snapshot_url)['status_code'] != 200:
-            time.sleep(self.wait_time)
-        if 'localhost' in dog._api_host:
-            snapshot_url = 'http://%s%s' % (dog.api_host, snapshot_url)
+        assert "event_query" not in snap
+        assert snap["metric_query"] == metric_query
+        snapshot_url = snap["snapshot_url"]
+        retry_count = 0
+        dog.Graph.status(snapshot_url)
+        while dog.Graph.status(snapshot_url)["status_code"] != 200 and retry_count < 10:
```

Comment:
if we setup `get_with_retry` to take in a failure scenario, we could de duplicate this logic

Reply:
done